src/dmd/expressionsem.d

```diff
@@ -2351,22 +2382,10 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             return;
         }
 
-        if (hasThis(sc))
+        if (Expression e = checkAliasThis(exp.loc, sc, exp.ident))
```

Comment:
I meant putting an `Expression e` outside of the if, assigning it, and then check for `if (e)`